eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -61,13 +84,230 @@ protected function configure()
+
+            if (!empty($processes)) {
+                sleep(1);
+            }
+        } while (!empty($processes));
+
+        $progress->finish();
+    }
+
+    /**
+     * @param DateTime $since
+     * @param string $select Field to select, one of  'c.id' and 'count(c.id)'
+     *
+     * @return \Doctrine\DBAL\Driver\Statement
+     */
+    private function getStatementContentSince(DateTime $since, $select = 'c.id')
+    {
+        $q = $this->connection->createQueryBuilder()
+            ->select($select)
+            ->from('ezcontentobject', 'c')
+            ->where('c.status = :status')->andWhere('c.modified >= :since')
+            ->orderBy('c.modified')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, PDO::PARAM_INT)
+            ->setParameter('since', $since->getTimestamp(), PDO::PARAM_INT);
+
+        return $q->execute();
+    }
+
+    /**
+     * @param string $select Field to select, one of  'c.id' and 'count(c.id)'
+     *
+     * @return \Doctrine\DBAL\Driver\Statement
+     */
+    private function getStatementContentAll($select = 'c.id')
+    {
+        $q = $this->connection->createQueryBuilder()
+            ->select($select)
+            ->from('ezcontentobject', 'c')
+            ->where('c.status = :status')
+            ->setParameter('status', ContentInfo::STATUS_PUBLISHED, PDO::PARAM_INT);
+
+        return $q->execute();
+    }
+
+    /**
+     * @param \Doctrine\DBAL\Driver\Statement $stmt
+     * @param int $iterationCount
+     *
+     * @return \Generator Return an array of arrays, each array contains content id's of $iterationCount.
+     */
+    private function fetchIteration(Statement $stmt, $iterationCount)
+    {
+        do {
+            $contentIds = [];
+            for ($i = 0; $i < $iterationCount; ++$i) {
+                if ($contentId = $stmt->fetch(PDO::FETCH_COLUMN)) {
+                    $contentIds[] = $contentId;
+                } else {
+                    break;
+                }
+            }
+
+            yield $contentIds;
+        } while (!empty($contentId));
+    }
+
+    /**
+     * @param array $contentIds
+     * @param bool $commit
+     *
+     * @return \Symfony\Component\Process\Process
+     */
+    private static function getPhpProcess(array $contentIds, $commit)
+    {
+        $phpFinder = new PhpExecutableFinder();
+        if (!$phpPath = $phpFinder->find()) {
+            throw new \RuntimeException('The php executable could not be found, add it to your PATH environment variable and try again');
+        }
+
+        $php = escapeshellarg($phpPath);
+        $console = escapeshellarg(file_exists('bin/console') ? 'bin/console' : 'app/console');
+        $cmd = 'ezplatform:reindex --content-ids=' . implode(',', $contentIds) . ($commit ? '' : ' --no-commit');
+
+        return new Process($php . ' ' . $console . ' ' . $cmd, null, null, null, null);
+    }
+
+    /**
+     * @return int
+     */
+    private function getNumberOfCPUCores()
```

Comment:
In general: please rely on `PHP_OS` constant (note: in PHP 7.2+ we will have `PHP_OS_FAMILY` with better naming).

Reply:
You mean instead of `DIRECTORY_SEPARATOR === '\\'` right, cause parsing OS names it worse then detecintg features of the OS. As for `DIRECTORY_SEPARATOR === '\\'`, this is how Symfony does it... avoids parsing it, and as long as we don't require PHP 7.2 it's probably thus the cleanest.

Reply:
Hmm, on second thought - good point, let's keep it like that. I'm actually not sure if for Windows it's gonna be `Windows`, `Win32` or `WINNT`, so even more in favor of your solution
+1